rmf_traffic_ros2/src/rmf_traffic_ros2/schedule/MirrorManager.cpp

```diff
@@ -303,6 +431,19 @@ class MirrorManager::Implementation
   {
     RCLCPP_INFO(node.get_logger(), "Handling fail over event for mirror");
 
+    // We need to validate that the replacement schedule node has our query
+    // correctly. This will be reset to false by one of:
+    // 1. The next set of queries being published, and our query being in it
+    //    correctly.
+    // 2. The query update topic being subscribed to, which indicates that
+    //    our query was not stored by the schedule node correctly and we have
+    //    re-registered.
+    // While true, all query updates received will be stashed in a list, and
+    // will be popped in FIFO order and processed when this is reset to false.
+    require_query_validation = true;
+    // The new schedule node will be one edition higher
+    ++expected_node_edition;
```

Comment:
I don't think we should have this here since it could create a conflict with the assignment [here](https://github.com/open-rmf/rmf_ros2/blob/f6f60726f69535c8c4d010ba8980f5223e62ccd6/rmf_traffic_ros2/src/rmf_traffic_ros2/schedule/MirrorManager.cpp#L275).

E.g. If we happen to process an update message from the new node with its new edition, then the `expected_node_edition` value will already be up to date. Then the fail over message may come in later and increment the value here, putting the `expected_node_edition` too high.

I think it would actually be fine to completely remove these lines here and leave it up to `update_handle` and the `queries_info_sub` callback to identify when we need to re-validate the query ID and `expected_node_edition`.

Reply:
Addressed by #87